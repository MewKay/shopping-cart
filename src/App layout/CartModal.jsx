import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import CartModalItem from "./CartModalItem";
import styles from "./CartModal.module.css";

const CartModal = ({
  showCartModal,
  totalQuantity,
  orderSubtotal,
  cart,
  handleHideCartModal,
}) => {
  const itemPlural = totalQuantity > 1 ? "s" : "";
  const modalOpenStyle = showCartModal ? styles["open"] : "";

  return (
    <div
      className={`
      ${styles["modal-overlay"]}
      ${modalOpenStyle}
      `}
      onClick={handleHideCartModal}
      aria-hidden={!showCartModal}
    >
      <div
        className={`
          ${styles["container"]}
          ${modalOpenStyle}
          `}
        onClick={(e) => e.stopPropagation()}
      >
        <button
          className={styles["close-modal-button"]}
          aria-label={"Close cart preview"}
          onClick={handleHideCartModal}
        >
          x
        </button>
        <div className={styles["cart-quantity-info"]}>
          Your cart contains {totalQuantity} item{itemPlural}
        </div>
        <div className={styles["cart-content"]}>
          <div className={styles["subtotal-info"]}>
            <p>Order Subtotal</p>
            <p className={styles["subtotal"]}>${orderSubtotal}</p>
          </div>
          <Link
            className={styles["cart-link"]}
            to={"/store/cart"}
            onClick={handleHideCartModal}
          >
            View or Edit Your Cart
          </Link>
          <ul className={styles["item-list"]}>
            {cart.map((item) => (
              <CartModalItem
                key={item.productDetails.id}
                image={item.productDetails.image}
                title={item.productDetails.title}
                price={item.productDetails.price}
                quantity={item.quantity}
              />
            ))}
          </ul>
        </div>
      </div>
    </div>
  );
};

CartModal.propTypes = {
  showCartModal: PropTypes.bool.isRequired,
  totalQuantity: PropTypes.number.isRequired,
  orderSubtotal: PropTypes.number.isRequired,
  cart: PropTypes.arrayOf(
    PropTypes.shape({
      productDetails: PropTypes.shape({
        id: PropTypes.number.isRequired,
        image: PropTypes.string,
        title: PropTypes.string.isRequired,
        price: PropTypes.number.isRequired,
      }),
      quantity: PropTypes.number.isRequired,
    })
  ),
  handleHideCartModal: PropTypes.func.isRequired,
};

export default CartModal;
