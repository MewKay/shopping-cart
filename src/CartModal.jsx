import { Link } from "react-router-dom";
import PropTypes from "prop-types";

const CartModal = ({ handleHideCartModal }) => {
  return (
    <div>
      <div>
        <button aria-label={"Close cart preview"} onClick={handleHideCartModal}>
          x
        </button>
        <div>Your cart contains 5 items</div>
        <div>
          <div>
            <p>Order Subtotal</p>
            <p>$659.00</p>
          </div>
          <Link to={"/store/cart"}>View or Edit Your Cart</Link>
        </div>
      </div>
    </div>
  );
};

CartModal.propTypes = {
  handleHideCartModal: PropTypes.func.isRequired,
};

export default CartModal;
