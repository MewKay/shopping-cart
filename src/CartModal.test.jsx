import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import CartModal from "./CartModal";
import { MemoryRouter } from "react-router-dom";

describe("CartModal component", () => {
  it("renders correctly", () => {
    const { container } = render(<CartModal />, { wrapper: MemoryRouter });

    expect(container).toMatchSnapshot();
  });
});
