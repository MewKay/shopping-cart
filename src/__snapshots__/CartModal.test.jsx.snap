// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CartModal component > renders correctly 1`] = `
<div>
  <div>
    <div>
      <button
        aria-label="Close cart preview"
      >
        x
      </button>
      <div>
        Your cart contains 5 items
      </div>
      <div>
        <div>
          <p>
            Order Subtotal
          </p>
          <p>
            $659.00
          </p>
        </div>
        <a
          href="/store/cart"
        >
          View or Edit Your Cart
        </a>
      </div>
    </div>
  </div>
</div>
`;
